diff --git a/module/bdev/aio/bdev_aio.c b/module/bdev/aio/bdev_aio.c
index 0b04cf052..fdc1697ea 100644
--- a/module/bdev/aio/bdev_aio.c
+++ b/module/bdev/aio/bdev_aio.c
@@ -128,7 +128,7 @@ bdev_aio_open(struct file_disk *disk)
 {
        int fd;

-       fd = open(disk->filename, O_RDWR | O_DIRECT);
+       fd = open(disk->filename, O_RDWR);
        if (fd < 0) {
                /* Try without O_DIRECT for non-disk files */
                fd = open(disk->filename, O_RDWR);
diff --git a/scripts/pkgdep/debian.sh b/scripts/pkgdep/debian.sh
index 7c7ba8ccf..c1bf36e2f 100755
--- a/scripts/pkgdep/debian.sh
+++ b/scripts/pkgdep/debian.sh
@@ -4,7 +4,7 @@ VERSION_ID_NUM=$(sed 's/\.//g' <<< $VERSION_ID)
 # Includes Ubuntu, Debian
 # Minimal install
 apt-get install -y gcc g++ make cmake libcunit1-dev libaio-dev libssl-dev libjson-c-dev libcmocka-dev \
-       uuid-dev libiscsi-dev python libncurses5-dev libncursesw5-dev python3-pip
+       uuid-dev libiscsi-dev python3 libncurses5-dev libncursesw5-dev python3-pip
 pip3 install ninja
 if ! pip3 install meson; then
        # After recent updates pip3 on ubuntu1604 provides meson version which requires python >= 3.6.
diff --git a/module/bdev/uring/bdev_uring.c b/module/bdev/uring/bdev_uring.c
index 51fab6384..89dab6605 100644
--- a/module/bdev/uring/bdev_uring.c
+++ b/module/bdev/uring/bdev_uring.c
@@ -100,7 +100,7 @@ bdev_uring_open(struct bdev_uring *bdev)
 {
        int fd;

-       fd = open(bdev->filename, O_RDWR | O_DIRECT | O_NOATIME);
+       fd = open(bdev->filename, O_RDWR | O_NOATIME);
        if (fd < 0) {
                /* Try without O_DIRECT for non-disk files */
                fd = open(bdev->filename, O_RDWR | O_NOATIME);
