diff --git a/module/bdev/aio/bdev_aio.c b/module/bdev/aio/bdev_aio.c
index 0b04cf052..fdc1697ea 100644
--- a/module/bdev/aio/bdev_aio.c
+++ b/module/bdev/aio/bdev_aio.c
@@ -128,7 +128,7 @@ bdev_aio_open(struct file_disk *disk)
 {
        int fd;

-       fd = open(disk->filename, O_RDWR | O_DIRECT);
+       fd = open(disk->filename, O_RDWR);
        if (fd < 0) {
                /* Try without O_DIRECT for non-disk files */
                fd = open(disk->filename, O_RDWR);