diff --git a/module/bdev/aio/bdev_aio.c b/module/bdev/aio/bdev_aio.c
index 7b33c71ad..62f6ca59f 100644
--- a/module/bdev/aio/bdev_aio.c
+++ b/module/bdev/aio/bdev_aio.c
@@ -113,7 +113,7 @@ bdev_aio_open(struct file_disk *disk)
 	int fd;
 	int io_flag = disk->readonly ? O_RDONLY : O_RDWR;
 
-	fd = open(disk->filename, io_flag | O_DIRECT);
+	fd = open(disk->filename, io_flag | O_SYNC);
 	if (fd < 0) {
 		/* Try without O_DIRECT for non-disk files */
 		fd = open(disk->filename, io_flag);
diff --git a/module/bdev/uring/bdev_uring.c b/module/bdev/uring/bdev_uring.c
index c000e7b75..b0a52467a 100644
--- a/module/bdev/uring/bdev_uring.c
+++ b/module/bdev/uring/bdev_uring.c
@@ -84,7 +84,7 @@ bdev_uring_open(struct bdev_uring *bdev)
 {
 	int fd;
 
-	fd = open(bdev->filename, O_RDWR | O_DIRECT | O_NOATIME);
+	fd = open(bdev->filename, O_RDWR | O_SYNC | O_NOATIME);
 	if (fd < 0) {
 		/* Try without O_DIRECT for non-disk files */
 		fd = open(bdev->filename, O_RDWR | O_NOATIME);
